src/vfio_device.rs

```diff
@@ -232,6 +236,7 @@ impl VfioContainer {
         Ok(())
     }
 
+    #[cfg(any(feature = "kvm", feature = "mshv"))]
```

Comment:
How about defining a dummy 'device_add_group()` and `device_del_group()` just to return success?